manticore/platforms/linux.py

```diff
@@ -250,6 +255,11 @@ def _transmit(self, buf):
             self.buffer.insert(0,c)
         return len(buf)
 
+    def sync(self):
+        raise BadFd("Invalid sync() operation on Socket")
```

Comment:
For this we were checking if we have a Socket and returning 
ret = -errno.EINVAL particularly for that. Will we ever do a sync on a socket?